packages/grid/src/vaadin-grid-mixin.js

```diff
@@ -812,6 +812,7 @@ export const GridMixin = (superClass) =>
     _updateRowStateParts(row, { expanded, selected, detailsOpened }) {
       updateBooleanRowStates(row, {
         expanded,
+        collapsed: this.__isRowExpandable(row),
```

Comment:
Another option is to add the property to the object returned by the `getRowModel` method, but I wasn't sure if we wanted to add the property there. It would be an API change, in a way.